universal-application-tool-0.0.1/app/views/admin/ProgramListView.java

```diff
@@ -0,0 +1,31 @@
+package views.admin;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.button;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.p;
+import static j2html.attributes.Attr.FORMACTION;
+
+import com.google.common.collect.ImmutableList;
+import play.twirl.api.Content;
+import services.program.ProgramDefinition;
+import views.BaseHtmlView;
+
+public final class ProgramListView extends BaseHtmlView {
+
+  public Content render(ImmutableList<ProgramDefinition> programs) {
+    return htmlContent(
+        body(
+            h1("Programs"),
+            div(
+                form(
+                    button("Add a Program")
+                        .attr(
+                            FORMACTION, controllers.admin.routes.AdminProgramController.newOne()))),
+            div(each(programs, program -> div(h2(program.name()), p(program.description()))))));
+  }
+}
```

Comment:
I think we need to split these out into sub-components as frequently as possible. When we start adding styles it should make things a lot simpler if the methods are small/distinct.

```
/** This should be in utils. */
Tag headerTag(String title) {
    return h1(title);
}

Tag renderLinkButton(String buttonText, Uri href) {
    return button(text).attr(onclick, String.format("window.location.href='%1s'", href))
}

Tag renderedProgramInfo(ProgramDefinition pd) {
    return div(
        h2(program.name()),
        p(program.description())
    );
}
```